zabbix/api.py

```diff
@@ -121,13 +121,22 @@ def do_request(self, method, params=None):
       z = ZabbixAPI()
       apiinfo = z.do_request('apiinfo.version')
     """
-    request_json = {
-      'jsonrpc':'2.0',
-      'method': method,
-      'params': params or {},
-      'auth': self.auth,
-      'id': '1',
-    }
+
+    if method = 'user.login':
```

Comment:
It should be different from api_version. But, because it will be much load to check version before request, we can just add `auth` if it's specified.

```python
if self.auth:
  request_json.update({'auth': self.auth})
```

i will test this with 2.2 zabbix